### PR TITLE
feat: custom multi label template

### PR DESCRIPTION
**Pull request by anjmao**

### CODE REVIEW

**Comment by anjmao** [comment]
https://github.com/ng-select/ng-select/issues/344

File: demo/app/examples/multi.component.ts [hunk 1]
```diff
@@ -76,6 +76,25 @@ import { DataService } from '../shared/data.service';
...
+        <label>Custom template for all selected items</label>
+        ---html,true
+        <ng-select
+            #api
+            [items]="githubUsers$ | async"
+            [multiple]="true"
+            bindLabel="login"
+            placeholder="Select items"
+            [(ngModel)]="selectedUsers">
+            <ng-template ng-multi-label-tmp let-item="item" let-items="items">
```
Comment (varnastadeus) [inline demo/app/examples/multi.component.ts]: remove `let-item="item"`

File: demo/app/examples/multi.component.ts [hunk 2]
```diff
@@ -76,6 +76,25 @@ import { DataService } from '../shared/data.service';
...
+
+        <label>Custom template for all selected items</label>
+        ---html,true
+        <ng-select
+            #api
+            [items]="githubUsers$ | async"
+            [multiple]="true"
+            bindLabel="login"
+            placeholder="Select items"
+            [(ngModel)]="selectedUsers">
```
Comment (varnastadeus) [inline demo/app/examples/multi.component.ts]: use another `ngModel`

File: demo/app/examples/multi.component.ts (hunk 1 above)
Comment (varnastadeus) [reply demo/app/examples/multi.component.ts]: screen shot 2018-03-14 at 10 33 47
make demo of something like this?

Comment (varnastadeus) [reply demo/app/examples/multi.component.ts]:
```html
<ng-template ng-multi-label-tmp let-items="items">
    <div class="ng-value" *ngFor="let item of (items ? items.slice(0,4): [])">
        <span class="ng-value-label"><img [src]="item.value.avatar_url" width="20px" height="20px"> {{item.value.login}}</span>
        <span class="ng-value-icon right" (click)="clear(item.value)" aria-hidden="true">×</span>
    </div>
    <div class="ng-value" *ngIf="items.length > 4" >
        <span class="ng-value-label">{{items.length - 4}} more...</span>
    </div>
</ng-template>
```

File: src/ng-select/ng-select.component.html [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
+                <ng-template
+                    [ngTemplateOutlet]="labelTemplate || defaultLabelTemplate"
+                    [ngTemplateOutletContext]="{ item: item.value, clear: clearItem, label: item.label }">
+                </ng-template>
+            </div>
+        </ng-container>
+
+        <ng-template *ngIf="multiLabelTemplate && selectedItems.length > 0"
+                [ngTemplateOutlet]="multiLabelTemplate"
+                [ngTemplateOutletContext]="{ items: selectedItems }">
```
Comment (varnastadeus) [inline src/ng-select/ng-select.component.html]: here you are exposing `selectedItems` which are internal, and user would need to know about `item.value`. 
Also exposing `clear: clearItem` would be useful

**Comment by varnastadeus** [comment]
also closes #336

File: demo/app/examples/multi.component.ts (hunk 1 above)
Comment (kkbalu) [reply demo/app/examples/multi.component.ts]: (click)="clear(item.value)" not working. whats the wrong in it.

Comment (kkbalu) [reply demo/app/examples/multi.component.ts]:
```suggestion
            <ng-template ng-multi-label-tmp let-items="items" let-clear="clear">
```

**Pull request merged**